sdk/src/Core/Internal/Utils/IPEndPointExtension.cs

```diff
@@ -95,6 +95,70 @@ public static bool TryParse(string input, out IPEndPoint endPoint)
                 return false;
             }
         }
+        
+        
+        /// <summary>
+        /// Tries to parse a string to <see cref="HostEndPoint"/>
+        /// </summary>
+        /// <param name="input">The input string.</param>
+        /// <param name="hostEndpoint">The parsed HostEndPoint</param>
+        /// <returns>true if <paramref name="input"/> converted successfully; otherwise, false.</returns>
+        public static bool TryParse(string input, out HostEndPoint hostEndpoint)
+        {
+            var entries = input.Split(':');
+            if (entries.Length != 2)
+            {
+                _logger.InfoFormat("Failed to parse HostEndPoint because input has not exactly two parts splitting by ':'. ({0})", input);
+                hostEndpoint = null;
+                return false;
+            }
+            if (!int.TryParse(entries[1], out var port))
+            {
+                _logger.InfoFormat("Failed to parse HostEndPoint because port is invalid. ({0})", input);
+                hostEndpoint = null;
+                return false;
+            }
+            if (port < 0 || 65535 < port)
+            {
+                _logger.InfoFormat("Failed to parse HostEndPoint because port is out of range. ({0})", input);
+                hostEndpoint = null;
+                return false;
+            }
+            
+            /*
+             * Almost anything can be a hostname which makes further validation here hard.
+             * Accept any string in entries[0] and let it fail in the DNS lookup instead.
+             */
+            
+            hostEndpoint = new HostEndPoint(entries[0], port);
+            _logger.InfoFormat("Using custom daemon address: {0}:{1}", hostEndpoint.Host, hostEndpoint.Port);
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to parse a string to <see cref="EndPoint"/>
+        /// </summary>
+        /// <param name="input">The input string.</param>
+        /// <param name="hostEndpoint">The parsed EndPoint</param>
+        /// <returns>true if <paramref name="input"/> converted successfully; otherwise, false.</returns>
+        public static bool TryParse(string input, out EndPoint endpoint)
+        {
+            if (TryParse(input, out IPEndPoint ipEndPoint))

```

Comment:
If `Hostname` is passed, this condition is false. However we get a log message `Failed to parse IPEndPoint because input is invalid.` because of this [check](https://github.com/aws/aws-xray-sdk-dotnet/blob/36c1c83e3d8fdf49763ffe15be84facd3d4cf45c/sdk/src/Core/Internal/Utils/IPEndPointExtension.cs#L54).
This log can be confusing for the customer who pass hostname. Instead, should we have helper method, say `IsIPAddress()` which validates whether its valid IP, if yes we call `(TryParse(input, out IPEndPoint ipEndPoint))` `else if (TryParse(input, out HostEndPoint hostEndPoint))`.

This approach helps in clean log messages.

Reply:
Addressed in commit 4e896fcf522dedaf5a0006fcbbe39c8217e25d2f.